### ISSUE TITLE
Simpler Annotation to make param be of type 'header'

### ISSUE THREAD

**Issue by magnusar** [swagger-api/swagger-core#696]
I don't know if this qualifies as an "issue" but I'll ask my question here anyway I suppose. We are using com.wordnik:swagger-jaxrs_2.11:1.3.10 in a Dropwizard project. We use Dropwizard's @Auth annotiation to inject authenticated principal objects into protected JAX-RS resource methods. Swagger doesn't seem to recognize the @Auth annotation so it's treated as a body param instead of an auth header. 

To solve this I wrote the following code:
```java
  @POST
  @Path("/mycall")
  @ApiOperation(value = "Does something...",
                notes = "Some description...",
                response = SomeClass.class)
  @ApiImplicitParams({ @ApiImplicitParam(name = "methodArgumentName",
                                         value = "Some info.",
                                         dataType = "string",
                                         paramType = "header") })
  public MyView viewRequest(@Auth AnotherClass methodArgumentName,
                                                @ApiParam(value = "bla bla", required = true) MoreClasses argumentName2)
```

The @ApiImplicitParams({...}) construct means a lot of code for every parameter we use with @Auth annotation. So my question(s) is:

1. Could you make Swagger aware of the @Auth annotation and automatically treat that as a header parameter instead of a body parameter?
2. Could you add some other parameter annotation that can be set directly on the parameter, for example like this: public MyView viewRequest(@Auth @HeaderParam AnotherClass methodArgumentName, ...
3. Could you extend the @ApiParam annotation to also have support for a paramType value? Like so: @ApiParam(value = "bla bla", required = true, paramType="header")
4. Is there any other way I can make the extra code smaller/simpler that's needed to define our @Auth annotated parameter as a header parameter?

Thanx in advance. (Ping @webron @fehguy )

**Comment by fehguy** [comment]
I have confirmed that option 2 works for any JAXRS resource.  Please reopen if needed.  Note, you can actually support (1) with the parameter extensions that swagger 1.5.x has introduced.  Just look at the swagger-jersey2-jaxrs module at how it is made available via SPI, and can be triggered when reading annotations.

**Comment by whisperwing** [comment]
@magnusar I know this is an old post but I'm running into the same issue and wonder what did you end up doing.

Adding @ApiImplicitParam(name = "methodArgumentName",...) didn't work for me, the auth parameter still shows up as a body parameter in the generated doc (using swagger version 1.5.1-M2). Did it work for you? What am I missing here?

**Comment by webron** [comment]
@whisperwing - there are ways around it, but you'd have to update to the latest version.

**Comment by whisperwing** [comment]
@webron OK I've updated to swagger-core 1.5.3 now. How do I work around this? Thanks!

**Comment by webron** [comment]
@whisperwing - what exactly do you need? To add the header parameter, remove the auth parameter from being a body parameter or both?

**Comment by whisperwing** [comment]
Both! I can add the header parameter explicitly with @ApiImplicitParam but I can't get rid of the body parameter

**Comment by webron** [comment]
You can, just add `@ApiParam(hidden=true)` next to it.

**Comment by whisperwing** [comment]
It worked, thanks Ron!

**Comment by webron** [comment]
That feature was added in 1.5.1 I think, which is why I asked you to upgrade. Glad it worked out.

**Comment by vsriram30** [comment]
@webron If we use @ApiParam(hidden=true), based on the source code comment, I get that it will be hidden from the list of parameters that are shown. But what if i want to enter the header param and make use of it? If it is hidden from the list of params, then I will not be able to enter that right?

Also looks like it is not in 1.5.1-M2 version which I am using because of [dropwizard-swagger](https://github.com/federecio/dropwizard-swagger/blob/master/pom.xml) module uses this version of swagger.

**Comment by fehguy** [comment]
it's hidden, so it won't be exposed to anyone.  A client can still make use of it, however the UI should not, by definition, show it.

**Comment by vsriram30** [comment]
@fehguy  Thanks for your reply. If the UI doesn't show it, then I will not be able to enter header params and try the API through swagger UI right? Is it still expected that APIs requiring headers should not be called via swagger?

**Comment by webron** [comment]
@vsriram30 what you're saying just doesn't make sense. Why would you hide a required parameter?

**Comment by vsriram30** [comment]
@webron Based on the above suggestions given by you to allow swagger pass header parameters that is what I got. I am new to swagger and might have misunderstood what you mentioned. I declared two params as header params in my method definition. It gets shown in swagger UI page as header param. But when I enter values for the header param and when I click Try it, the call is not directed to my implemented method. But when I call the same API through curl command with header added with -H, the call is going to the implemented API method. Hence I was not sure why when called via swagger UI, the call is not reaching my implementation.
